tools/test-proxy/Azure.Sdk.Tools.TestProxy/RecordingHandler.cs

```diff
@@ -985,11 +998,177 @@ public void SetDefaultExtensions(string recordingId = null)
+                    // new HeaderRegexSanitizer("x-ms-encryption-key"), dup
+                    new HeaderRegexSanitizer("x-ms-rename-source"),
+                    new HeaderRegexSanitizer("x-ms-file-rename-source"),
+                    new HeaderRegexSanitizer("x-ms-copy-source"),
+                    new HeaderRegexSanitizer("x-ms-copy-source-authorization"),
+                    new HeaderRegexSanitizer("x-ms-file-rename-source-authorization"),
+                    new HeaderRegexSanitizer("x-ms-encryption-key-sha256"),
+                    new BodyKeySanitizer("$..uploadUrl"),
+                    new BodyKeySanitizer("$..logLink"),
+                    new HeaderRegexSanitizer("aeg-sas-token"),
+                    new HeaderRegexSanitizer("aeg-sas-key"),
+                    new HeaderRegexSanitizer("aeg-channel-name"),
+                    new BodyKeySanitizer("$..storageContainerUri"),
+                    new BodyKeySanitizer("$..storageContainerReadListSas"),
+                    new BodyKeySanitizer("$..storageContainerWriteSas"),
+                    new BodyRegexSanitizer(regex: "token=(?<token>[^&]+)($|&)", groupForReplace: "token"),
+                    new BodyRegexSanitizer(regex: "-----BEGIN PRIVATE KEY-----\\n(.+\\n)*-----END PRIVATE KEY-----\\n"),
+                    new BodyKeySanitizer("$..primaryMasterKey"),
+                    new BodyKeySanitizer("$..primaryReadonlyMasterKey"),
+                    new BodyKeySanitizer("$..secondaryMasterKey"),
+                    new BodyKeySanitizer("$..secondaryReadonlyMasterKey"),
+                    new BodyKeySanitizer("$..password"),
+                    new BodyKeySanitizer("$..certificatePassword"),
+                    new BodyKeySanitizer("$..clientSecret"),
+                    new BodyKeySanitizer("$..keyVaultClientSecret"),
+                    new BodyKeySanitizer("$..accountKey"),
+                    new BodyKeySanitizer("$..authHeader"),
+                    new BodyKeySanitizer("$..httpHeader"),
+                    new BodyKeySanitizer("$..encryptedCredential"),
+                    new BodyKeySanitizer("$..appkey"),
+                    new BodyKeySanitizer("$..functionKey"),
+                    new BodyKeySanitizer("$..atlasKafkaPrimaryEndpoint"),
+                    new BodyKeySanitizer("$..atlasKafkaSecondaryEndpoint"),
+                    new BodyKeySanitizer("$..certificatePassword"),
+                    new BodyKeySanitizer("$..storageAccountPrimaryKey"),
+                    new BodyKeySanitizer("$..privateKey"),
+                    new BodyKeySanitizer("$..fencingClientPassword"),
+                    new BodyKeySanitizer("$..acrToken"),
+                    new BodyKeySanitizer("$..scriptUrlSasToken"),
+                    new BodyKeySanitizer("$..refresh_token"),
+                    new BodyRegexSanitizer(regex: "(?<=<UserDelegationKey>).*?(?:<Value>)(.*)(?:</Value>)"),
```

Comment:
I think we need to use GroupForReplace for these - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs#L160

Reply:
Done.